core/domain/classifier_services.py

```diff
@@ -270,80 +276,69 @@ def get_classifier_training_job_by_id(job_id):
-    classifier training job domain object.
+def create_classifier_training_job(algorithm_id, interaction_id, exp_id,
+                                   exp_version, state_name, training_data,
+                                   status):
+    """Creates a ClassifierTrainingJobModel in data store.
 
     Args:
-        classifier_training_job: ClassifierTrainingJob. Domain object for the
-            classifier training job.
+        algorithm_id: str. ID of the algorithm used to generate the model.
+        interaction_id: str. ID of the interaction to which the algorithm
+            belongs.
+        exp_id: str. ID of the exploration.
+        exp_version: int. The exploration version at the time
+            this training job was created.
+        state_name: str. The name of the state to which the classifier
+            belongs.
+        training_data: dict. The data used in training phase.
+        status: str. The status of the training job (NEW by default).
 
     Returns:
         job_id: str. ID of the classifier training job.
     """
+    dummy_classifier_training_job = classifier_domain.ClassifierTrainingJob(
+        'job_id_dummy', algorithm_id, interaction_id, exp_id, exp_version,
+        state_name, status, training_data)
+    dummy_classifier_training_job.validate()
     job_id = classifier_models.ClassifierTrainingJobModel.create(
-        classifier_training_job.algorithm_id, classifier_training_job.exp_id,
-        classifier_training_job.exp_version,
-        classifier_training_job.training_data,
-        classifier_training_job.state_name)
+        algorithm_id, interaction_id, exp_id, exp_version, training_data,
+        state_name)
     return job_id
 
 
-def _update_classifier_training_job(classifier_training_job_model, status):
-    """Updates classifier training job model in the datastore given a
-    classifier training job domain object.
+def _update_classifier_training_job_status(job_id, status):
+    """Checks for the existence of the model and then updates it.
 
     Args:
-        classifier_training_job_model: ClassifierTrainingJobModel. Classifier
-            training job model instance in datastore.
-        status: The status of the job.
+        job_id: str. ID of the ClassifierTrainingJob domain object.
+        status: str. The status to which the job needs to be updated.
 
-    Note: All of the properties of a classifier training job are immutable,
-        except for status.
+    Raises:
+        Exception. The ClassifierTrainingJobModel corresponding to the job_id
+            of the ClassifierTrainingJob does not exist.
     """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id, strict=False))
+    if not classifier_training_job_model:
+        raise Exception(
+            'The ClassifierTrainingJobModel corresponding to the job_id of the'
+            'ClassifierTrainingJob does not exist.')
+
```

Comment:
I think you should define a constant that's something like ALLOWED_TRAINING_JOB_STATUS_CHANGES that maps the current status to the allowed "next statuses". Before doing anything here, check that the status transition is valid, and raise an exception if it isn't. Then, if any status change flows through this function in the future, validity is guaranteed.